### PR TITLE
LoKI: Fix things in script builder codes

### PR DESCRIPTION
**Pull request by umithardal**
This PR fix some PEP issues, adds headers and make some other corrections to the codes that handle script generation and table manipulation for LoKI.

On the side note:
@mattclarke and @ebadkamil 

If one chooses `SIMULTANEOUS` as trans order in settings in `Prototype` tab, Python throws an error concerning positional arguments (required 2 but given 3). This only happens for `SIMULTANEOUS`.

### CODE REVIEW

**Comment by mattclarke** [comment]
This looks good mostly, thanks for tidying. 👍 

I would prefer not to have the staticmethods though, so can you remove those please.
It is better to add them later when the API has settled down and we know they are truly static and as a lot of this is in flux now is not the best time.

I think PyCharm is overzealous on warning on this, so I would recommend switching that warning off.

**Comment by ebadkamil** [comment]
I can't reproduce the BUG related to `SIMULTANEOUS`

**Comment by umithardal** [comment]
> This looks good mostly, thanks for tidying. 👍
> 
> I would prefer not to have the staticmethods though, so can you remove those please.
> It is better to add them later when the API has settled down and we know they are truly static and as a lot of this is in flux now is not the best time.
> 
> I think PyCharm is overzealous on warning on this, so I would recommend switching that warning off.

Well, in this case `self` adds to the bug. For example now you have an error it says `3 expected given 4` for `Simultaneous` while in reality of course these numbers are not correct (as far as I understood). As this goes to main, I would have make it proper. Anyways. I have cleaned the decorators and pushed.

File: nicos_ess/loki/gui/script_generator.py [hunk 1]
```diff
@@ -98,8 +130,8 @@ def generate_script(self, labeled_data, trans_duration_type,
 
 
 class Simultaneous:
-    def generate_script(self, labeled_data, trans_duration_type,
-                        sans_duration_type):
+    @staticmethod
+    def generate_script(labeled_data, sans_duration_type):
```
Comment (ebadkamil) [inline nicos_ess/loki/gui/script_generator.py]: Let's keep the `trans_duration_type` although it is not used, to keep the signature of method intact.

**Comment by umithardal** [comment]
> I can't reproduce the BUG related to `SIMULTANEOUS`

It may be os dependent?

File: nicos_ess/loki/gui/script_generator.py (hunk 1 above)
Comment (umithardal) [reply nicos_ess/loki/gui/script_generator.py]: Do we have future plans for that value?

File: nicos_ess/loki/gui/loki_scriptbuilder.py [hunk 2]
```diff
@@ -148,16 +176,16 @@ def on_loadTableButton_clicked(self):
...
     def _fill_table(self, headers, data):
         # corresponding indices of elements in headers_from_file list to headers
-        indices = [i for i, e in enumerate(self.columns_in_order)
-                   if e in headers]
+        indices = [index for index, element in
+                   enumerate(self.columns_in_order) if element in headers]
 
         table_data = []
-        for idx, row in enumerate(data):
+        for index, row in enumerate(data):
```
Comment (ebadkamil) [inline nicos_ess/loki/gui/loki_scriptbuilder.py:188]: Also get rid of `enumerate`  here as `index`  is no longer used.

**Comment by umithardal** [comment]
@ebadkamil This is the full traceback:

`nicos_ess/loki/gui/loki_scriptbuilder.py", line 410, in on_generateScriptButton_clicked
    self.comboSansDurationType.currentText())
TypeError: generate_script() takes 3 positional arguments but 4 were given
[15:06:57] gui       : ERROR: unhandled exception in QT callback - TypeError - generate_script() takes 3 positional arguments but 4 were given
`

File: nicos_ess/loki/gui/script_generator.py (hunk 1 above)
Comment (umithardal) [reply nicos_ess/loki/gui/script_generator.py]: Adding that argument fixes the issue on simultaneous. I do not know if this is a good way. However I pushed the code.

**Comment by mattclarke** [comment]
I cannot reproduce the bug either. 
The signature for simulatenous should be:

**Comment by umithardal** [comment]
> I cannot reproduce the bug either.
> The signature for simulatenous should be:
> 
> ```
> class Simultaneous:
>     def generate_script(self, labeled_data, trans_duration_type,
>                         sans_duration_type):
> ```
> 
> Even though trans_duration_type isn't used it needs to be there to keep the APIs consistent.
> Is it possible it got deleted on your working copy?

I fixed it by re-adding the value there. Sorry for the false alarm. I answered that to Ebad probably you did not see it.

**Comment by mattclarke** [comment]
Great, thanks @umithardal

**Review by mattclarke — APPROVED** [review]

**Comment by ebadkamil** [comment]
Thanks. LGTM

**Pull request merged**